src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,30 @@
 package mate.academy;
 
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final Set<EventListener> listeners = ConcurrentHashMap.newKeySet();
```

Comment:
Using ConcurrentHashMap.newKeySet() is a good choice for thread-safe set operations. However, ensure that the rest of the class is designed to handle concurrent access if necessary.